datashuttle/utils_mod/utils.py

```diff
@@ -313,23 +258,55 @@ def process_names(
     return prefixed_names
 
 
-def update_ses_names_with_datetime(names: list):
+def update_names_with_datetime(names: list):
     """
-    Replate @DATE and @DATETIME flag with date and datetime respectively.
-    Currently formats time as XXhXXh.
+    Replace @DATE and @DATETIME flag with date and datetime respectively.
 
-    Dont format datetime directly so we can keep timezone aware.
+    Format using key-value pair for bids, i.e. date-20221223_time-
     """
-    date = str(datetime.datetime.now().date())
-    date = date.replace("-", "")
-    time_ = datetime.datetime.now().time().strftime("%Hh%Mm")
-    datetime_ = f"{date}-{time_}"
-
-    for i, val in enumerate(names):
-        if "@DATETIME" in val:
-            names[i] = val.replace("@DATETIME", datetime_)
-        elif "@DATE" in val:
-            names[i] = val.replace("@DATE", date)
+    date = str(datetime.datetime.now().date().strftime("%Y%m%d"))
+    format_date = f"date-{date}"
+
+    time_ = datetime.datetime.now().time().strftime("%H%M%S")
+    format_time = f"time-{time_}"
+
+    for i, name in enumerate(names):
+
+        if "@DATETIME" in name:  # must come first
```

Comment:
Is it worth storing these specific strings (e.g. `@DATETIME`) somewhere that can be imported, so we don't need to find & replace if we ever change them?

Reply:
Nice idea I think so, this is now implemented ([link to commit](https://github.com/neuroinformatics-unit/datashuttle/pull/66/commits/b1399dc78e3f4b21a4899ce2f74535a69301aa14))